lib/internal/blob.js

```diff
@@ -56,29 +65,54 @@ const {
 const kHandle = Symbol('kHandle');
 const kType = Symbol('kType');
 const kLength = Symbol('kLength');
+const kArrayBufferPromise = Symbol('kArrayBufferPromise');
 
 const disallowedTypeCharacters = /[^\u{0020}-\u{007E}]/u;
 
-let Buffer;
+let ReadableStream;
+let URL;
+let EOL;
+
+const enc = new TextEncoder();
+
+// Yes, lazy loading is annoying but because of circular
+// references between the url, internal/blob, and buffer
+// modules, lazy loading here makes sure that things work.
+
+function lazyURL(id) {
+  URL ??= require('url').URL;
+  return new URL(id);
+}
+
+function lazyReadableStream(options) {
+  ReadableStream ??=
+    require('internal/webstreams/readablestream').ReadableStream;
+  return new ReadableStream(options);
+}
 
-function lazyBuffer() {
-  if (Buffer === undefined)
-    Buffer = require('buffer').Buffer;
-  return Buffer;
+// TODO(@jasnell): This is annoying but this has to be lazy because
+// requiring the 'os' module too early causes building Node.js to
+// fail with an unknown reference failure.
+function lazyEOL() {
+  EOL ??= require('os').EOL;
+  return EOL;
 }
 
 function isBlob(object) {
   return object?.[kHandle] !== undefined;
 }
 
-function getSource(source, encoding) {
+function getSource(source, endings) {
   if (isBlob(source))
     return [source.size, source[kHandle]];
 
   if (isAnyArrayBuffer(source)) {
     source = new Uint8Array(source);
   } else if (!isArrayBufferView(source)) {
-    source = lazyBuffer().from(`${source}`, encoding);
+    source = `${source}`;
+    if (endings === 'native')
+      source = source.replaceAll(/(\n|\r\n)/g, lazyEOL());
```

Comment:
```suggestion
      source = RegExpPrototypeSymbolReplace(/(\n|\r\n)/g, source, lazyEOL());
```

Reply:
I don't believe there's any reason to use a capture group here.

```suggestion
      source = source.replaceAll(/\n|\r\n/g, lazyEOL());
```

Or if using @aduh95's suggestion:

```suggestion
      source = RegExpPrototypeSymbolReplace(/\n|\r\n/g, source, lazyEOL());
```